threshsign/src/eddsa/EdDSAMultisigVerifier.cpp

```diff
@@ -48,9 +48,8 @@ size_t EdDSASignatureAccumulator::getFullSignedData(char *outThreshSig, int thre
   ConcordAssertGE(static_cast<uint64_t>(threshSigLen),
                   static_cast<unsigned long>(signatures_.size()) * sizeof(SingleEdDSASignature));
   size_t offset = 0;
-  for (auto &[id, sig] : signatures_) {
-    UNUSED(id);
-    std::memcpy(outThreshSig + offset, &sig, sizeof(SingleEdDSASignature));
+  for (const auto &signature : signatures_) {
```

Comment:
I wont comment more about this repeating issue, but I think parameter pack is the right choice, and you know already the way how to handle the UNUSED from my other comments. I would bring it back to how it was (with the const ref).

Reply:
gcc/g++ compilation fails for `[[maybe_unused]]`. Hence, not retaining parameter pack.

Reply:
Retaining UNUSED macro for below reason:
- `[[maybe_unused]]` could have been used instead of `UNUSED()`, but compilation with gcc/g++ fails (however clang++ compilation passes).
The g++ compiler version needs an upgrade to atleast 10 or 11. Current g++ version is v7.5.0. In current g++ version, `[[maybe_unused]]` is not recognised in cases such as range-based for loop (e.g. `for ([[maybe_unused]] const auto &[id, sig] : signatures_) {}`). In this example, if `id` is unused, g++ compiler throws an compilation error: "`error: unused variable 'id' [-Werror=unused-variable]`"